src/Orleans.Core/Serialization/OrleansJsonSerializer.cs

```diff
@@ -71,10 +72,10 @@ public static JsonSerializerSettings UpdateSerializerSettings(JsonSerializerSett
             bool useFullAssemblyNames = config.GetBoolProperty(UseFullAssemblyNamesProperty, false);
             bool indentJson = config.GetBoolProperty(IndentJsonProperty, false);
             TypeNameHandling typeNameHandling = config.GetEnumProperty(TypeNameHandlingProperty, settings.TypeNameHandling);
-            return UpdateSerializerSettings(settings, useFullAssemblyNames, indentJson, typeNameHandling);
+            return UpdateSerializerSettings(settings, useFullAssemblyNames, indentJson, typeNameHandling, null);
         }
 
-        public static JsonSerializerSettings UpdateSerializerSettings(JsonSerializerSettings settings, bool useFullAssemblyNames, bool indentJson, TypeNameHandling? typeNameHandling)
+        public static JsonSerializerSettings UpdateSerializerSettings(JsonSerializerSettings settings, bool useFullAssemblyNames, bool indentJson, TypeNameHandling? typeNameHandling, Action<JsonSerializerSettings> configureSettings)
```

Comment:
Since this is a public method, it's best to not break back-compat.

The configureSettings parameter does not need to be passed in to this method: callers can each call it.
eg:
```C#
this.JsonSettings = OrleansJsonSerializer.UpdateSerializerSettings(OrleansJsonSerializer.GetDefaultSerializerSettings(this.typeResolver, this.grainFactory), this.options.UseFullAssemblyNames, this.options.IndentJson, this.options.TypeNameHandling);

this.options.ConfigureJsonSerializerSettings?.Invoke(this.JsonSettings);
```

Reply:
Ok, never thought of that. Will make the change.